indexer/services/ender/src/lib/block-processor.ts

```diff
@@ -232,8 +232,10 @@ export class BlockProcessor {
         ...this.block.events[eventProto.blockEventIndex],
         subtype: SKIPPED_EVENT_SUBTYPE,
       };
-      // Set handlers to empty array if event is to be skipped.
-      handlers = [];
+      // Set handlers to empty array if a non-stateful-order event is to be skipped.
+      if (eventProto.type !== DydxIndexerSubtypes.STATEFUL_ORDER) {
```

Comment:
This is also an empty pattern - we adding some obscure conditional chains that makes this code hard to reason about.

Seems like the validator should able to differentiate between `shouldExcludeEventForSql` and `shouldExcludEventForHandler`.